app/src/common/shared/org/mozilla/vrbrowser/telemetry/GleanMetricsService.java

```diff
@@ -0,0 +1,63 @@
+package org.mozilla.vrbrowser.telemetry;
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.annotation.UiThread;
+
+import org.mozilla.vrbrowser.browser.SettingsStore;
+import org.mozilla.vrbrowser.utils.DeviceType;
+import org.mozilla.vrbrowser.utils.SystemUtils;
+import org.mozilla.vrbrowser.BuildConfig;
+import org.mozilla.vrbrowser.GleanMetrics.Metrics;
+
+import mozilla.components.service.glean.Glean;
+import mozilla.components.service.glean.config.Configuration;
+
+
+public class GleanMetricsService {
+
+    private final static String APP_NAME = "FirefoxReality";
+    private static boolean initialized = false;
+    private final static String LOGTAG = SystemUtils.createLogtag(GleanMetricsService.class);
+    private static Context context = null;
+
+    // We should call this at the application initial stage.
+    public static void init(Context aContext) {
+        if (initialized)
+            return;
+
+        context = aContext;
+        initialized = true;
+
+        final boolean telemetryEnabled = SettingsStore.getInstance(aContext).isTelemetryEnabled();
+        Glean.INSTANCE.setUploadEnabled(telemetryEnabled);
+        Configuration config = new Configuration(Configuration.DEFAULT_TELEMETRY_ENDPOINT,
+                                                 BuildConfig.BUILD_TYPE);
+        Glean.INSTANCE.initialize(aContext, config);
+
+        if (!Glean.INSTANCE.isInitialized()) {
+            Log.d(LOGTAG, "Glean doesn't be initialized yet.");
+        }
+
+    }
+
+    // It would be called when users turn on/off the setting of telemetry.
+    // e.g., SettingsStore.getInstance(context).setTelemetryEnabled();
+    @UiThread
```

Comment:
Is there any particular reason why this is marked as `@UiThread`? It should not be needed for Glean.

Reply:
Telemetry service seems to need, so I just follow it. Good to know Glean doesn't need to run in UIThread.